memory/src/main/scala/filodb.memory/MemFactory.scala

```diff
@@ -280,6 +280,19 @@ class BlockMemFactory(blockStore: BlockManager,
       metaAddr = blk.allocMetadata(metaSize)
       if (metaAddr != 0) metadataWriter(metaAddr)
     }
+
+    if (metaAddr == 0) {
```

Comment:
I'm not sure we should do this.  The reason is, that metadata is supposed to be co-located with chunks in the same block, for the express purpose of properly being able to notify the TSPartition owners of their memory being recycled/reclaimed.  If it is located in a different block, then the ability to notify on free properly is compromised.  Granted, not having metadata allocated also compromises that.   I'm thinking, we need to actually address why the metadata could not be allocated in the first place.  It should not happen if the `allocMetadata` has already taken cared of capacity for the metadata, but maybe there are edge conditions we aren't thinking of.   Actually, what we really probably need is an atomic "allocate metadata and space for all chunks".

Reply:
The edge case is quite simple. Start a metaspan and then end it. This used to cause a segv, but then an exception was thrown instead. It doesn't really fix the problem. I can perhaps change where the exception is thrown, but what is the contract supposed to be?

Reply:
Ok, not that simple. The metaspan needs to be started and ended when there's no room left.

Reply:
Hmmm.  So ideally what we want is:
1) If the endMetaspan was called with no previous chunks allocated in that span, go ahead and flip to a new block
2) however if chunks were already allocated in the current block, one needs to allocate metadata in the current block and not the next block

Reply:
I think the problem lies in the fact that metadata size is both pinned to the BlockMemFactory (in the ctor) and the caller can override this. By eliminating the caller override, the start metaspan can check early and move to another block.